src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Android/AndroidCertificatePal.cs

```diff
@@ -417,5 +454,15 @@ private void EnsureCertificateData()
             Debug.Assert(!_cert.IsInvalid);
             _certData = new CertificateData(RawData);
         }
+
+        private ICertificatePal CopyWithPrivateKeyHandle(SafeKeyHandle? privateKey)
+        {
+            // Add a global reference to the underlying cert object.
+            SafeX509Handle handle = new SafeX509Handle(Interop.JObjectLifetime.NewGlobalReference(Handle));
+            if (privateKey == null)
+                return new AndroidCertificatePal(handle);
```

Comment:
I don't think null is expected here... is it?

Reply:
I'm using this function for `FromOtherCert` as well. I can separate it, since I'd agree `CopyWithPrivateKeyHandle` expecting a null private key is weird/non-intuitive.